File: backend/schemas/admin_allowed_song.py
from pydantic import BaseModel, ConfigDict
from typing import Optional


class AdminAllowedSongBase(BaseModel):
    admin_user_id: int
    song_id: int


class AdminAllowedSongCreate(AdminAllowedSongBase):
    pass


class AdminAllowedSongResponse(AdminAllowedSongBase):
    # Include song details in response
    song_title: Optional[str] = None
    artist: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/apiTypes/adminAllowedSong.ts
// Admin Allowed Song types matching backend schemas

export interface AdminAllowedSong {
    admin_user_id: number;
    song_id: number;
    song_title?: string;
    artist?: string;
}

export interface AdminAllowedSongCreate {
    admin_user_id: number;
    song_id: number;
}
